osu.Game/Overlays/News/Sidebar/YearsPanel.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using System.Collections.Generic;
+using osu.Game.Graphics;
+using osu.Framework.Bindables;
+using osu.Game.Online.API.Requests.Responses;
+using System;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.News.Sidebar
+{
+    public class YearsPanel : CompositeDrawable
```

Comment:
I started to refactor this to make some sense, but then realised it should be a simple `FillFlow` not a `GridContainer`. I'll leave this to you else this is going to become my own PR.

Reply:
Using a grid isn't entirely unwarranted, as web *does* use a 4-column grid layout there, though. However as I said earlier regarding this exact same point, I can't really bring myself to care which one it should be in the end.

Reply:
The four column layout can already be forced by setting the size of the columns to relative width 0.25 or similar.